pkg/cluster/sync_test.go

```diff
@@ -177,6 +216,41 @@ func TestConnectionPoolerSynchronization(t *testing.T) {
 			defaultInstances: 1,
 			check:            objectsAreDeleted,
 		},
+		{
+			subTest: "delete only master if not needed",
+			oldSpec: &acidv1.Postgresql{
+				Spec: acidv1.PostgresSpec{
+					ConnectionPooler: &acidv1.ConnectionPooler{},
+				},
+			},
+			newSpec: &acidv1.Postgresql{
+				Spec: acidv1.PostgresSpec{
+					EnableReplicaConnectionPooler: boolToPointer(true),
+				},
+			},
+			cluster:          clusterMock,
+			defaultImage:     "pooler:1.0",
+			defaultInstances: 1,
+			check:            OnlyMasterDeleted,
+		},
+		{
+			subTest: "delete only replica if not needed",
```

Comment:
It's probably necessary to test other transitions, e.g. from only master pooler to both, from both to one etc.